go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -1028,8 +1029,11 @@ func (throttler *Throttler) generateTabletProbeFunction(scope base.Scope, tmClie
 			return metricsWithError(fmt.Errorf("gRPC error accessing tablet %v. Err=%v", probe.Alias, gRPCErr))
 		}
 		throttleMetric.Value = resp.Value
+		if resp.ResponseCode == tabletmanagerdatapb.CheckThrottlerResponseCode_INTERNAL_ERROR {
+			throttleMetric.Err = fmt.Errorf("response code: %d", resp.ResponseCode)
+		}
 		if resp.StatusCode == http.StatusInternalServerError {
-			throttleMetric.Err = fmt.Errorf("Status code: %d", resp.StatusCode)
+			throttleMetric.Err = fmt.Errorf("status code: %d", resp.StatusCode)
 		}
```

Comment:
Should this be an `else if`? I wouldn't think we'd want the http code to override the response code for the error, but maybe we do.

Reply:
The two cpde blocks will behave identically. I intentionally separated to two separate `if`s because for backwards compatibility sake, I must keep the old one, so adding as a separate block is "clean" diff-wise. In `v22` we will remove that old one, and again the diff will be clean. WDYT?

Reply:
Sure, not a big deal at all.